### ISSUE TITLE
plot_settings kwarg in GenericMap gets passed to NDData

### ISSUE THREAD

**Issue by wtbarnes** [sunpy/sunpy#1968]
[GenericMap allows for passing kwargs like `plot_settings` so that, e.g. the default colormap can be changed.](https://github.com/sunpy/sunpy/blob/master/sunpy/map/mapbase.py#L161) It looks like only the `**kwargs` (which presumably NDData should be fine with?) should be passed to NDData, but plot settings is getting passed into the NDData constructor as well and this throws an unexpected keyword argument error.

## MWE
```Python
import sunpy.map
import matplotlib.colors
import seaborn
sdo_map = sunpy.map.Map('/path/to/an_aia_fits_file.fits') #valid SDO/AIA map
plot_settings =  {
            'cmap':matplotlib.colors.ListedColormap(seaborn.color_palette('coolwarm',n_colors=1000)),
            'norm':matplotlib.colors.SymLogNorm(10,vmin=-1e8,vmax=1e8)
}
new_map = sunpy.map.GenericMap(sdo_map.data,sdo_map.meta,plot_settings=plot_settings)
```
throws the error,
```Python
TypeError                                 Traceback (most recent call last)
<ipython-input-15-9b1e8d91517e> in <module>()
      3             'norm':matplotlib.colors.SymLogNorm(10,vmin=-1e8,vmax=1e8)
      4         }
----> 5 new_map = sunpy.map.GenericMap(sdo_map.data,sdo_map.meta,plot_settings=plot_settings)

/opt/anaconda3/envs/synthesizar_refactor/lib/python3.5/site-packages/sunpy/map/mapbase.py in __init__(self, data, header, **kwargs)
    172     def __init__(self, data, header, **kwargs):
    173 
--> 174         super(GenericMap, self).__init__(data, meta=header, **kwargs)
    175 
    176         # Correct possibly missing meta keywords

/opt/anaconda3/envs/synthesizar_refactor/lib/python3.5/site-packages/sunpy/map/nddata_compat.py in __init__(self, *arg, **kwd)
     25         def __init__(self, *arg, **kwd):
     26             # Initialize with the parent...
---> 27             super(NDDataCompat, self).__init__(*arg, **kwd)
     28 
     29             # ...then reset uncertainty to force it to go through the

TypeError: __init__() got an unexpected keyword argument 'plot_settings'
```

**Comment by Cadair** [comment]
This is the same error as what is causing memmap to fail. We need to only pass explicit kwargs to the NNData.__init__ method. I think someone was asking about fixing this in IRC recently.

**Comment by Cadair** [comment]
To add a little direction for anyone who wants to pick this up:

This line: https://github.com/sunpy/sunpy/blob/master/sunpy/map/mapbase.py#L163 needs to have `**kwargs` removed, and instead all supported keyword arguments to `astropy.nddata.NDData` need to be extracted from the `kwargs` dict and passed into the `__init__` call explicitly. (It's probably best to use `dict.get` to do that while giving default values).

**Comment by saurvs** [comment]
I'm not sure this is an bug with the current version of Sunpy (I wasn't able to reproduce it with the most recent checkout of the repo).

In the traceback given above, `GenericMap.__init__` has the signature `(self, data, header, **kwargs)`, whereas https://github.com/sunpy/sunpy/commit/06baddcb2a21d29ccbb289826686cbccf2ce69d8 explicitly added the `plot_settings=plot_settings` argument to it. So any code passing `plot_settings` to the prior version of `__init__` would have it end up inside `kwargs`, causing the `TypeError`.

**Comment by Cadair** [comment]
@saurvs interesting. Even if it isn't fixing this bug, my proposed solution was also designed to fix this: https://github.com/sunpy/sunpy/issues/1745#issuecomment-266889218

**Comment by wtbarnes** [comment]
@saurvs Yes I think you're correct. The traceback comes from 0.7.3. I didn't notice the lack of `plot_settings` in the `GenericMap` signature above. 

Either way, since 0.7.3 is the latest stable release, this bug fix should still be backported. The separate `plot_settings` kwarg could be abandoned altogether (unless people have a preference for it in the docstring of `GenericMap`) in favor of just linting `**kwargs` before it gets passed to `NDData`. As @Cadair pointed out, this needs to be done for the `memmap` kwarg anyway.

**Comment by Cadair** [comment]
@wtbarnes plot settings being an argument to `GenericMap` is a post 0.7 feature(?), I don't think it is supported at all.

We should keep keyword arguments explicit where possible, noting that for `GenericMap` we want it to have `**kwargs` and silently ignore extra args to make things easier in the factory class.

**Comment by wtbarnes** [comment]
@Cadair ah ok fair enough. I think my main confusion was looking at the current master branch and not realizing that the `plot_settings` kwarg was implemented after 0.7.3. I think this issue can be closed then.

**Comment by Cadair** [comment]
We should open one for the memmap issue. Or re-title the other one I linked. Will do later.
